java/java.source.base/src/org/netbeans/modules/java/source/indexing/JavaCustomIndexer.java

```diff
@@ -147,6 +151,76 @@ protected void index(final Iterable<? extends Indexable> files, final Context co
                 JavaIndex.LOG.fine("Ignoring request with no root"); //NOI18N
                 return;
             }
+            BinaryForSourceQuery.Result2 binRes = BinaryForSourceQuery.findBinaryRoots2(root.toURL());
+            if (binRes.preferBinaries()) {
+                final URL[] binaryRoots = binRes.getRoots();
+                final FileObject[] binaryRootsFo = new FileObject[binaryRoots.length];
+                Long newestFile = null;
+                int at = 0;
+                for (URL u : binaryRoots) {
+                    FileObject ufo = URLMapper.findFileObject(u);
+                    if (ufo == null) {
+                        newestFile = null;
+                        break;
+                    }
+                    binaryRootsFo[at++] = ufo;
+                    Enumeration<? extends FileObject> en = ufo.getChildren(true);
+                    while (en.hasMoreElements()) {
+                        FileObject ch = en.nextElement();
+                        long modified = ch.lastModified().getTime();
+                        if (newestFile == null || newestFile < modified) {
+                            newestFile = modified;
+                        }
+                    }
+                }
+
+                boolean binariesAreNewer = true;
+                for (Indexable index : files) {
+                    if (!binariesAreNewer) {
```

Comment:
Can be removed, no?
The `binariesAreNewer = true` is followed by `break`.

Reply:
Done in 64160669b763